### PR TITLE
Mri violations upgrades

### PR DESCRIPTION
**Pull request by tedstrauss**

### CODE REVIEW

File: SQL/2015-03-20_MRIViolationsResolved.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+CREATE TABLE IF NOT EXISTS `violations_resolved` (
+  `ID` bigint(20) NOT NULL AUTO_INCREMENT,
+  `hash` varchar(255) NOT NULL,
+  `ExtID` bigint(20) NOT NULL,
+  `TypeTable` varchar(255) DEFAULT NULL,
+  `User` varchar(255) DEFAULT NULL,
+  `ChangeDate` datetime DEFAULT NULL,
+  `Resolved` enum('0','1','2','3','4','5','6','7','8','9') DEFAULT '0',
```
Comment (driusan) [inline SQL/2015-03-20_MRIViolationsResolved.sql]: This enum should be more descriptive instead of depending on magic numbers

File: SQL/2015-03-20_MRIViolationsResolved.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  PRIMARY KEY (`ID`)
+) ENGINE=InnoDB AUTO_INCREMENT=46 DEFAULT CHARSET=utf8;
```
Comment (driusan) [inline SQL/2015-03-20_MRIViolationsResolved.sql]: I think AUTO_INCREMENT should start at 1

File: modules/mri_violations/php/NDB_Menu_Filter_Form_mri_violations.class.inc [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+        if (Utility::isErrorX($user)) {
+            return PEAR::raiseError("User Error: ".$user->getMessage());
+        }
+        foreach ($values['resolvable'] AS $key=>$val) {
+            $hash = $key;
+            $row = $DB->pselectRow(
+                    "SELECT * FROM violations_resolved
+                    WHERE hash = :hash", array('hash' => $hash));
+            
+            //if row is found it means that resolve status was previously assigned,
```
Comment (driusan) [inline modules/mri_violations/php/NDB_Menu_Filter_Form_mri_violations.class.inc]: Can these be combined into a single $DB->replace() call to simplify this if the hash was made the primary key?

File: modules/mri_violations/php/NDB_Menu_Filter_Form_resolved_violations.class.inc [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+ * scan, etc) and link to the appropriate module to further investigate
+ * why a scan was excluded by the imaging pipeline scripts.
+ *
+ * PHP Version 5
+ *
+ * @category MRI
+ * @package  Main
+ * @author   Dave MacFarlane <driusan@bic.mni.mcgill.ca>
+ * @license  GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
```
Comment (driusan) [inline modules/mri_violations/php/NDB_Menu_Filter_Form_resolved_violations.class.inc]: This header is the same as the previous page, one of them is wrong (I think this one, since I didn't write it..)

File: modules/mri_violations/php/NDB_Menu_Filter_Form_resolved_violations.class.inc [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+     *
+     * @return boolean true if the user is permitted to see violated scans
+     */
+    function _hasAccess()
+    {
+        // create user object
+        $user =& User::singleton();
+        if (Utility::isErrorX($user)) {
+            return PEAR::raiseError("User Error: ".$user->getMessage());
+        }
```
Comment (driusan) [inline modules/mri_violations/php/NDB_Menu_Filter_Form_resolved_violations.class.inc]: Take out the ::isErrorX calls since the functions throw exceptions now.

File: php/libraries/NDB_Caller.class.inc [hunk 6]
```diff
@@ -148,7 +149,13 @@ class NDB_Caller extends PEAR
 
                 $this->type = 'form';
                 return $html;
-            }
+            } /*else{ //if($this->existsAndRequire("NDB_Menu_Filter_Form_$test_name.class.inc")){
```
Comment (driusan) [inline php/libraries/NDB_Caller.class.inc]: Can you remove this? We shouldn't have commented out code in the codebase, and shouldn't have HTML in the PHP..

File: tools/install.sh [hunk 7]
```diff
@@ -328,8 +328,8 @@ done;
 
 
 echo ""
-echo "Attempting to create and grant privileges to MySQL user '$mysqluser'@'localhost' ..."
-echo "GRANT UPDATE,INSERT,SELECT,DELETE ON $mysqldb.* TO '$mysqluser'@'localhost' IDENTIFIED BY '$mysqlpass' WITH GRANT OPTION" | mysql $mysqldb -h$mysqlhost --user=$mysqlrootuser --password="$mysqlrootpass" -A > /dev/null 2>&1
+echo "Attempting to create and grant privileges to MySQL user '$mysqluser'@'$mysqlhost' ..."
+echo "GRANT UPDATE,INSERT,SELECT,DELETE ON $mysqldb.* TO '$mysqluser'@'$mysqlhost' IDENTIFIED BY '$mysqlpass' WITH GRANT OPTION" | mysql $mysqldb -h$mysqlhost --user=$mysqlrootuser --password="$mysqlrootpass" -A > /dev/null 2>&1
```
Comment (driusan) [inline tools/install.sh]: This looks like it's supposed to be a different pull request, can you separate it and take it out of this one?

(Also, I don't think this change will work.. the @ part is the host that the user is connecting from, not the MySQL host.)

File: php/libraries/LorisForm.class.inc [hunk 8]
```diff
@@ -373,7 +372,7 @@ class LorisForm
                 $optionKey = (string) $optionKey;
             }
 
-            if ($optionKey === $this->getValue($el['name'])) {
+            if ($optionKey === $el['selected']) {
```
Comment (driusan) [inline php/libraries/LorisForm.class.inc]: I think there's a different pull request for this change, so it shouldn't be part of this one (unless it depends on the change, in which case we should wait until that one finishes getting reviewed/merged..)

File: modules/mri_violations/php/NDB_Menu_Filter_Form_resolved_violations.class.inc (hunk 4 above)
Comment (driusan) [inline modules/mri_violations/php/NDB_Menu_Filter_Form_resolved_violations.class.inc]: I'm not really clear on what the difference is between the two files now that I've looked at them both, I'll re-review after the comment for one of them is updated so I have a better conceptual understanding of the difference..

File: SQL/2015-03-20_MRIViolationsResolved.sql (hunk 1 above)
Comment (tedstrauss) [inline SQL/2015-03-20_MRIViolationsResolved.sql]: ya good point, i can update those

**Comment by tedstrauss** [comment]
todo:
- remove empty graph frame
- remove unused ajax
- delete code in comments
- Not Resolved tab > remove V-R select
- Resolved tab > tweak V-R select

File: tools/install.sh (hunk 7 above)
Comment (tedstrauss) [inline tools/install.sh]: I would like to remove that but the merged commit can't be undone. so if this breaks something i'll have to remove it in a separate PR after this one is pulled.

File: composer.json [hunk 9]
```diff
@@ -13,7 +13,8 @@
         "pear-pear.php.net/Pager" : "~2.4",
         "pear-pear.php.net/Benchmark" : "~1.2",
         "pear-pear.php.net/HTML_QuickForm" : "~3.2",
-        "pear-pear.php.net/Mail" : "~1.2"
+        "pear-pear.php.net/Mail" : "~1.2",
+        "cviebrock/discourse-php" : "~0.9.1"
```
Comment (driusan) [inline composer.json]: I don't think this change is supposed to be in this pull request.

File: modules/mri_violations/php/NDB_Menu_Filter_Form_mri_violations.class.inc [hunk 10]
```diff
@@ -0,0 +1,441 @@
...
+            return PEAR::raiseError("User Error: ".$user->getMessage());
+        }
+        $this->tpl_data['violated_scans_modifications'] 
+            = $user->hasPermission('violated_scans_edit');
+        return ($user->hasPermission('violated_scans_view_allsites'));
+    }
+
+    function _process($values)
+    {
+        print_r($values);
```
Comment (driusan) [inline modules/mri_violations/php/NDB_Menu_Filter_Form_mri_violations.class.inc]: Seems to be a leftover debugging statement

File: modules/mri_violations/php/NDB_Menu_Filter_Form_mri_violations.class.inc [hunk 11]
```diff
@@ -0,0 +1,441 @@
...
+                if($val=='unresolved'){
+                    continue;
+                }
+                $newlyResolved = array();
+                $newlyResolved['hash'] = $key;                
+                $newlyResolved['Resolved'] = (string)$val;
+                $newlyResolved['User'] = $user->getUsername();
+                $newlyResolved['ChangeDate'] = date("Y-m-d H:i:s");
+
+                echo "key $key\n";
```
Comment (driusan) [inline modules/mri_violations/php/NDB_Menu_Filter_Form_mri_violations.class.inc]: more debugging statements

**Comment by tedstrauss** [comment]
@driusan have all your concerns been answered for this PR?

File: php/libraries/LorisForm.class.inc (hunk 8 above)
Comment (tedstrauss) [inline php/libraries/LorisForm.class.inc]: ok trying to remove this commit

**Comment by driusan** [comment]
I just had a quick look, and the default schema hasn't been updated to correspond to the SQL patch that's included, and it looks like the selected/LorisForm changes are still in it (and don't include the update that you fixed yesterday.)

(It also can't be merged because the button is greyed out and it needs to be rebased...)

**Comment by driusan** [comment]
Can you add a change to this to trigger Travis? It seems to be working on new changes again.
